core/src/tile/tileManager.cpp

```diff
@@ -339,38 +371,54 @@ void TileManager::enqueueTask(const TileSet& tileSet, const TileID& tileID,
                                [](auto& distance, auto& other){
                                    return distance < std::get<0>(other);
                                });
+
     m_loadTasks.insert(it, std::make_tuple(distance, &tileSet, &tileID));
 }
 
 void TileManager::loadTiles() {
 
     for (auto& item : m_loadTasks) {
-        auto& id = *std::get<2>(item);
+
+        auto id = *std::get<2>(item);
         auto& tileSet = *std::get<1>(item);
-        auto it = tileSet.tiles.find(id);
-        if (it == tileSet.tiles.end()) { continue; }
 
-        auto& tile = it->second;
-        auto& source = tileSet.source;
+        auto tileIt = tileSet.tiles.find(id);
+        if (tileIt == tileSet.tiles.end()) {
```

Comment:
I am not sure if this is required here. According to me there is no case when a `tile` in a `loadTask` will be removed from the `tileSet` (based on how `updateTileSet` works).